packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1868,6 +1930,26 @@ class ThemeData with Diagnosticable {
     return Brightness.dark;
   }
 
+  /// Linearly interpolate between two [extensions].
+  /// 
+  /// Includes all theme extensions in [a] and [b].
+  ///
+  /// {@macro dart.ui.shadow.lerp}
+  static Map<Object, ThemeExtension<Object>> _lerpThemeExtensions(ThemeData a, ThemeData b, double t) {
+    // Lerp in place.
+    a.extensions.map((Object id, ThemeExtension<Object> extensionA) {
+        final ThemeExtension<Object>? extensionB = b.extensions[id];
+        return MapEntry<Object, ThemeExtension<Object>>(id, extensionA.lerp(extensionB, t));
```

Comment:
ThemeData.lerp creates new object. 
And extensionA.lerp(...) - does not modify extensionA, and dart Map.map - returns new map.
So this to my understanding is actually noop?

As well set manipulation in this case does not give much performance boost, so maybe:


final newExtensions = a.extensions.map(...);
for (final MapEntry<Object, ThemeExtension<Object>> entry in b.extensions.entries) {
   if(!newExtensions.containsKey(entry.key)) newExtensions.addEntry(entry);
}

Reply:
Ah true, fixed.